pkg/skaffold/initializer/build/util.go

```diff
@@ -97,8 +98,6 @@ func Artifacts(pairs []BuilderImagePair) []*latest.Artifact {
 			artifact.Workspace = workspace
 		}
 
-		pair.Builder.UpdateArtifact(artifact)
```

Comment:
My only concern here is that lines 94–98 may update the artifact's `Workspace`, which could be relevant to `UpdateArtifact`.  I suppose since that workspace is from the same `InitBuilder` then `ArtifactType()` should have that same information.

Would it be worth moving the Artifact creation entirely into the InitBuilders?

Reply:
I which case do you think it would be useful for `UpdateArtifact` to know the `Workspace`? I wanted to simplify the code but I'm ok to go the opposite way if there's a concrete need for it

Reply:
I think it's a definite improvement.  If we need the workspace, we can amend the function to pass in the workspace.